### ISSUE TITLE
Option to block all Pleroma servers from federating

### ISSUE THREAD

**Issue by a-dows** [hometown-fork/hometown#1176]
### Pitch

Implement an admin UI feature that would prevent any and all Pleroma servers from federating with a hometown server.

### Motivation

As an admin I've continued to see abuses that seem to be entirely localized to servers running Pleroma. Things like a feature that lets you follow a user's unlocked posts even if they deny your follow request, and just today, a functional @ everyone tag that spams tens of thousands of external users at once.

These servers are overwhelmingly likely to be single-user instances and there are so many of them it's extremely difficult to discover and suspend them one at a time.

I have heard from several server admins that if they could, they would block all Pleroma servers. The benefits outweigh any benign Pleromans that might be there.

**Comment by lawremipsum** [comment]
I'd like something like this, though I think it would be useful for it to include

-  A chronological list of newly discovered instances blocked by default with an option to allow (could also be used for allow-list mode for all newly discovered and disallowed instances)—while the % of pleroma users that aren't abusive seems to be relatively small, they are out there. But I agree the default action that requires no intervention should be suspend.

- a setting to allow-list instances that would be blocked but have existing mutual follower relationships, with an option to suspend, which would send them to the other list.

**Comment by misterducky** [comment]
Glad someone actually is submitting this.

**Comment by nevillepark** [comment]
I would also appreciate this.

**Comment by dariusk** [comment]
Just looking at this now. I agree that I end up blocking Pleroma servers more than any other federated software. I think that instead of making this a feature to ban Pleroma, I could come up with a feature that lets you filter any server based on, say, what software they self-identify as using. That way if I wanted to ban everyone running anything that isn't Hometown... well, I could do that! (Not that I would but you know. I like having the option.)

**Comment by a-dows** [comment]
@dariusk I support this implementation!

**Comment by dariusk** [comment]
Closing this in favor of #1198.
